src/borg/repository.py

```diff
@@ -872,7 +879,7 @@ def report_error(msg):
                 continue
             try:
                 objects = list(self.io.iter_objects(segment))
-            except IntegrityError as err:
+            except (IntegrityError, LoggedIO.AtticSegment) as err:
                 report_error(str(err))
                 objects = []
                 if repair:
```

Comment:
hmm...

i see 2 options:

- catch the AtticSegment exception separately and re-raise it as AtticRepository exception (== "check --repair" is not "upgrade from attic")
- actually change "check --repair" to be the "upgrade from attic" - there is a ticket open about the upgrade only being able to work locally, but if we would do it via "borg check" in the repository code, it could work remotely also.

to not have scope-creep in here, maybe do the first here and we can discuss doing the second later.

@enkore ^